corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -172,7 +178,23 @@ hqDefine("cloudcare/js/formplayer/menus/api", function () {
                 });
 
                 var callStartTime = performance.now();
-                menus.fetch($.extend(true, {}, options)).always(function () {
+                const updateRequest = menus.fetch($.extend(true, {}, options));
+
+                if (route.startsWith("navigate_menu")) {
+                    if (!_.isEqual(params.selections, currentSelections)) {
```

Comment:
With SSCS there can be a query and an entities request that are going on at the same time. This makes sure that we only cancel the request if we are navigating away.

Reply:
I don't follow - doesn't checking the route confirm that we're navigating away?

Reply:
route is just `navigate_menu`, `get_details` and such. The bug was caused by two parallel `navigate_menu` requests. One for the search and one for the navigation in the bread crumbs. The difference between the two would be the selection.

Reply:
I still don't quite get it. Is this accurate?

Bug
1. User clicks search
2. That calls `queryFormplayer` with the `navigate_menu` route and sends a request to formplayer
3. User clicks breadcrumbs, say to go back to the previous menu
4. That calls `queryFormplayer` with the `navigate_menu` route and sends a second request to formplayer
5. The second request finishes. The user is navigated to their menu. The important part is that the formplayer code handling that second request finished before the formplayer code handling the first request.
6. The first request errors on the formplayer side, because...the app is now in a different state where that first navigation request isn't legit? (This part I'm fuzzy on, since formplayer is generally stateless.)

Solution
1. User clicks search
2. That calls `queryFormplayer` with the `navigate_menu` route and sends a request to formplayer
   * That hits the new code, which saves the given selections as `currentSelections`. 
3. User clicks breadcrumbs, say to go back to the previous menu
4. That calls `queryFormplayer` with the `navigate_menu` route and sends a second request to formplayer
   * That hits the new code, which sees that the given selections are different than `currentSelections`. It updates `currentSelections`. It looks at `ongoingRequests`, which contains the first request, and aborts it. (This is my previous question - why does it matter that the selections are different? If we had two requests with the same selections, say the user double-clicked a breadcrumb, wouldn't we still be fine to abort one of them?)
   * Formplayer stops processing the first request and sends back an error response, which is caught by the new code in the error handler.
5. The second request finishes on the formplayer side, sends back a response, and the user is navigated to their menu.

Reply:
Bug 6. is wrong. It does not error out but set the selection back to menu, form.
7. Navigate back to case list. Now you get an error caused by the wrong selection in the session storage.

Solution 4. See "With SSCS there can be a query and an entities request that are going on at the same time. This makes sure that we only cancel the request if we are navigating away." above. Those two request would have the same route (navigate_menu) and selection. Hence the checks.

Reply:
ohh, got it, thank you.

I feel like it's a little bit weird that `ongoingRequests` is a list when it seems this is to address a very specific situation with exactly two requests being triggered, but that's a nitpick.